### PR TITLE
cgifsave: add support for interlaced GIF write

### PR DESCRIPTION
**Pull request by dloebl**
cgif >= [v0.3.0](https://github.com/dloebl/cgif/releases/tag/V0.3.0) supports writing interlaced (progressive) GIFs. Add support for interlaced GIF write in libvips.
**Note:** As of now, cgif requires more memory for writing interlaced GIFs (tracked by https://github.com/dloebl/cgif/issues/52).

### CODE REVIEW

File: libvips/foreign/cgifsave.c [hunk 1]
```diff
@@ -601,6 +602,16 @@ vips_foreign_save_cgif_write_frame( VipsForeignSaveCgif *cgif )
...
 	}
 
+	/* Write an interlaced GIF, if requested.
+	*/
+	if( cgif->interlace ) {
+#ifdef HAVE_CGIF_FRAME_ATTR_INTERLACED
+		frame_config.attrFlags |= CGIF_FRAME_ATTR_INTERLACED;
+#else /*!HAVE_CGIF_FRAME_ATTR_INTERLACED*/
+		vips_error( class->nickname, "%s", _( "cgif >= v0.3.0 required for interlaced GIF write" ) );
+#endif /*HAVE_CGIF_FRAME_ATTR_INTERLACED*/
```
Comment (jcupitt) [inline libvips/foreign/cgifsave.c:612]: This should probably be a warning, or if this is an error, `return -1;` after.

**Comment by jcupitt** [comment]
Looks nice! Maybe add a test as well?

File: libvips/foreign/cgifsave.c (hunk 1 above)
Comment (dloebl) [reply libvips/foreign/cgifsave.c:612]: > This should probably be a warning, or if this is an error, `return -1;` after.

Sounds good. I changed it to be just a warning.

**Comment by jcupitt** [comment]
Great!

**Comment by jcupitt** [comment]
Ooof, I forgot, there ought to be a one-line note in CHANGELOG. I'll add it.

**Pull request merged**